### PR TITLE
EL-280 - (Version 7.0.0) - Support for CM API v3.3

### PR DESCRIPTION
**Pull request by ijer**
Upgrades to Createsend API v3.3 which includes new breaking changes

* New client tags() returns the list of tags 
* Client draftCampaigns() now return Tags
* Client scheduledCampaigns() now returns Tags
* Revamped client sentCampaigns() includes these improvements: 
  * The method returns a PagedResult of SentCampaign.
  * Entries can be fetched using the Page parameter.
  * Up to 1000 entries are returned per method call. A different amount can be returned by setting the PageSize value.
  * Campaigns can be sorted by their Sent Date using OrderDirection
  * Campaigns can be filtered by Sent Date using sentFromDate and sentToDate in YYYY-MM-DD format
  * Campaigns can be filtered by their tags.
* Campaign summary() now returns Name

### CODE REVIEW

File: src/com/createsend/Clients.java [hunk 1]
```diff
@@ -116,16 +119,95 @@ public String create(Client client) throws CreateSendException {
...
+     * @param orderDirection The direction to order results by.
+     *                       The value can be ASC or DESC
+     *                       Use <code>null</code> for the default (DESC)
+     * @return The paged campaigns returned by the api call
+     * @throws CreateSendException Thrown when the API responds with a HTTP Status >= 400
+     * @see <a href="http://www.campaignmonitor.com/api/clients/#getting_client_campaigns" target="_blank">
+     * Getting sent campaigns</a>
+     */
+    public PagedResult<SentCampaign> sentCampaigns(
+            Date sentFromDate, Date sentToDate, String[] tags,
```
Comment (YahyaZ) [inline src/com/createsend/Clients.java:184]: Should `tags` be a string array? Or do we keep it consistent with the API endpoint and just have it be comma delimited (This is what we do in the C#, Python, Ruby PRs)

**Review by YahyaZ — COMMENTED** [review]
Nice work @ijer one question

Also we'll need to update the readme examples too

**Review by henrys-cm — APPROVED** [review]
LGTM -- looking at the extra commits from Yahya and looks good.

File: samples/com/createsend/samples/SampleRunner.java [hunk 2]
```diff
@@ -550,7 +550,7 @@ private static void runClientMethods() throws AssertException, CreateSendExcepti
                 Arrays.deepToString(clientAPI.sentCampaigns().Results));
 
         try {
-            String[] tags = {"tag1", "tag2"};
+            String[] tags = "tag1, tag2";
```
Comment (henrys-cm) [inline samples/com/createsend/samples/SampleRunner.java:553]: I think you meant this to be String rather than String[]?

**Comment by henrys-cm** [comment]
LGTM -- had able to run samples locally too for client for these changes

File: HISTORY.md [hunk 3]
```diff
@@ -1,5 +1,20 @@
 # createsend-java history
 
+## v7.0.0 - 19 November 2021
+
+* Upgrades to Createsend API v3.3 which includes new breaking changes
+* New client tags() returns the list of tags 
+* Client draftCampaigns() now return Tags
+* Client scheduledCampaigns() now returns Tags
+* Revamped client sentCampaigns() includes these improvements:
```
[the remark this answers is not in the document]
Comment (ijer) [reply HISTORY.md]: Good call

File: src/com/createsend/Campaigns.java [hunk 4]
```diff
@@ -450,7 +450,7 @@ public ListsAndSegments listsAndSegments() throws CreateSendException {
     /**
      * Gets a paged list of spam complaints for the specified campaign
      * @param spamComplaintsFrom The date to start getting spam complaints from. This field is required
-     * @param page The page number or results to get. Use <code>null</code> for the default (page=1)
+     * @param page the page number of results to get. Use <code>null</code> for the default (page=1)
```
[the remark this answers is not in the document]
Comment (ijer) [reply src/com/createsend/Campaigns.java]: No good reason.  Typos

File: src/com/createsend/Clients.java (hunk 1 above)
Comment (ijer) [reply src/com/createsend/Clients.java]: Interesting.  In my opinion, the wrapper should offer a string array.  I believe that's the correct mapping between the comma-delimited parameter of the API and the client application.

**Pull request merged**